pew/pew.py

```diff
@@ -184,7 +184,10 @@ def _detect_shell():
         if 'CMDER_ROOT' in os.environ:
             shell = 'Cmder'
         elif windows:
-            shell = psutil.Process(os.getpid()).parent().parent().name()
+            try:
+                _, shell = shellingham.detect_shell()
+            except shellingham.ShellDetectionFailure:
+                shell = os.environ['COMSPEC']
```

Comment:
Thanks for your PR!

I wasn't aware of the COMSPEC environment variable, so that's :+1: 

but 

1- the check should be moved above, together with the other checks for env variables (the "fancier" logic provided by psutil or shellingam should be a fallback), if we have an explicit SHELL (or if the user is using cmder, or if COMSPEC is provided) that's a more explicit bit of info about what shell the user actually wants to run
2- This can fail with KeyError, the other code that access os.environ is a bit more resilient

Reply:
COMSPEC is *always* set on Windows, but doesn't necessarily reflect the shell the user is currently running (it's the "default" shell, basically). So we can't check it in preference to the shellingham call, it needs to be left as a fallback. Having said that, handling (pathological) cases where `COMSPEC` isn't set seems worthwhile. I'd suggest `os.environ.get('COMSPEC', 'cmd.exe')`

Reply:
Good point

SHELL is also similar from that point of view, but you won't find that variable always set